File: Objects/TimeBlock.cs
using System;
using System.Drawing;
using System.Xml.Serialization;

namespace GanttChart
{
    public class TimeBlock
    {
        public TimeBlock()
        {
            this.Hatch = false;
            this.Clickable = true;
            this.IsVisible = true;
        }

        public TimeBlock(string text)
        {
            this.Text = text;
            this.Hatch = false;
            this.Clickable = true;
            this.IsVisible = true;
        }

        public TimeBlock(string text, DateTime startTime, DateTime endTime)
        {
            this.Text = text;
            this.StartTime = startTime;
            this.EndTime = endTime;
            this.Hatch = false;
            this.Clickable = true;
            this.IsVisible = true;
        }

        public virtual string Text { get; set; }
        public virtual DateTime StartTime { get; set; }
        public virtual DateTime EndTime { get; set; }
        public virtual Color Color { get; set; }
        public virtual bool Hatch { get; set; }
        public virtual bool Clickable { get; set; }
        public virtual Rectangle Rect { get; set; }
        public virtual bool IsVisible { get; set; }
    }
}

File: GanttChart/Objects/TimeBlock.vb
Imports System.Drawing

Namespace GanttChart
    Public Class TimeBlock
        Public Sub New()
            Hatch = False
            Clickable = True
            IsVisible = True
        End Sub

        Public Sub New(text As String)
            Me.Text = text
            Hatch = False
            Clickable = True
            IsVisible = True
        End Sub

        Public Sub New(text As String, startTime As Date, endTime As Date)
            Me.Text = text
            Me.StartTime = startTime
            Me.EndTime = endTime
            Hatch = False
            Clickable = True
            IsVisible = True
        End Sub

        Public Overridable Property Text As String
        Public Overridable Property StartTime As Date
        Public Overridable Property EndTime As Date
        Public Overridable Property Color As Color
        Public Overridable Property Hatch As Boolean
        Public Overridable Property Clickable As Boolean
        Public Overridable Property Rect As Rectangle
        Public Overridable Property IsVisible As Boolean
    End Class
End Namespace
